qcore/im.py

```diff
@@ -129,7 +129,10 @@ class IM:
     def __post_init__(self):
         if not isinstance(self.name, IMEnum):
             self.name = IMEnum[self.name]
-        if self.component is not None and not isinstance(self.component, constants.Components):
+
+        if self.component is not None and not isinstance(
+            self.component, constants.Components
+        ):
             constants.Components.from_str(self.component)
```

Comment:
While you are editing this are you able to fix my bug?

`self.component = constants.Components.from_str(self.component)`

Reply:
Done.